src/TraceEvent/Parsers/LinuxKernelEventParser.cs

```diff
@@ -0,0 +1,215 @@
+        public enum Keywords : long
+        {
+            Process = 0x1,
+        };
+
+        public LinuxKernelEventParser(TraceEventSource source) : base(source) {}
+
+        public event Action<ProcessStartTraceData> ProcessStart
+        {
+            add
+            {
+                source.RegisterEventTemplate(new ProcessStartTraceData(value, 1, 1, "Process", ProcessTaskGuid, 1, "Start", ProviderGuid, ProviderName));
+            }
+            remove
+            {
+                source.UnregisterEventTemplate(value, 1, ProcessTaskGuid);
+            }
+        }
+        public event Action<ProcessStopTraceData> ProcessStop
+        {
+            add
+            {
+                source.RegisterEventTemplate(new ProcessStopTraceData(value, 2, 1, "Process", ProcessTaskGuid, 2, "Stop", ProviderGuid, ProviderName));
+            }
+            remove
+            {
+                source.UnregisterEventTemplate(value, 2, ProcessTaskGuid);
+            }
+        }
+
+        #region private
+        protected override string GetProviderName() { return ProviderName; }
+
+        static private volatile TraceEvent[] s_templates;
+
+/*        static private ProcessStartTraceData ProcessStartTemplate(Action<ProcessStartTraceData> action)
+        {                  // action, eventid, taskid, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName
+            return new ProcessStartTraceData(action, 1, 1, "Process", Guid.Empty, 1, "Start", ProviderGuid, ProviderName);
+        }
+        static private ProcessStopTraceData ProcessStopTemplate(Action<ProcessStopTraceData> action)
+        {                  // action, eventid, taskid, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName
+            return new ProcessStopTraceData(action, 2, 1, "Process", Guid.Empty, 2, "Stop", ProviderGuid, ProviderName);
+        }*/
+
+        protected internal override void EnumerateTemplates(Func<string, string, EventFilterResponse> eventsToObserve, Action<TraceEvent> callback)
+        {
+            if (s_templates == null)
+            {
+                var templates = new TraceEvent[2];
+                templates[0] = new ProcessStartTraceData(null, 1, 1, "Process", ProcessTaskGuid, 1, "Start", ProviderGuid, ProviderName);
+                templates[1] = new ProcessStopTraceData(null, 2, 1, "Process", ProcessTaskGuid, 2, "Stop", ProviderGuid, ProviderName);
+                s_templates = templates;
+            }
+            foreach (var template in s_templates)
+                if (eventsToObserve == null || eventsToObserve(template.ProviderName, template.EventName) == EventFilterResponse.AcceptEvent)
+                    callback(template);
+        }
+
+        #endregion
+    }
+}
+
+namespace Microsoft.Diagnostics.Tracing.Parsers.LinuxKernel
+{
+    public sealed class ProcessStartTraceData : TraceEvent
+    {
+/*      public int ProcessID { get { return GetInt32At(0); } }
```

Comment:
I'm assuming that these are commented out because these properties exist in the `TraceEvent` base class?  You can probably get away with not having these fields here at all because they are present in the event context associated with each event if you use `lttng add-context`.  Are there other fields beyond these that we want to capture, such as command line?

Reply:
Yes these properties already exist in the base class and sometimes the parsing doesn't work properly. I plan to add other fields after this prototype works.